File: java/52.java
class Solution {
    private int count(int i, int s, boolean c[], boolean r[], boolean l[]) {
        if (i == s)
            return 1;
        int q = 0;
        for (int j = 0; j < s; j++)
            if (!(c[j] || r[i + j] || l[s - i + j])) {
                c[j] = true;
                r[i + j] = true;
                l[s - i + j] = true;
                q += count(i + 1, s, c, r, l);
                c[j] = false;
                r[i + j] = false;
                l[s - i + j] = false;
            }
        return q;
    }

    public int totalNQueens(int n) {
        return count(0, n, new boolean[n], new boolean[2 * n], new boolean[2 * n]);
    }
}

File: python3/52.py
class Solution:
    def totalNQueens(self, n: int) -> int:
        def count(i, s, c, r, l):
            if i == s:
                return 1
            q = 0
            for j in range(s):
                if not (c[j] or r[i + j] or l[s - i + j]):
                    c[j] = True
                    r[i + j] = True
                    l[s - i + j] = True
                    q += count(i + 1, s, c, r, l)
                    c[j] = False
                    r[i + j] = False
                    l[s - i + j] = False
            return q

        return count(0, n, [False] * n, [False] * 2 * n, [False] * 2 * n)
